pyqtorch/utils.py

```diff
@@ -218,9 +218,7 @@ def is_diag(H: Tensor, atol: Tensor = ATOL) -> bool:
     Returns:
         True if diagonal, else False.
     """
-    m = H.shape[0]
-    p, q = H.stride()
-    offdiag_view = torch.as_strided(H[:, 1:], (m - 1, m), (p + q, q))
```

Comment:
Why did you need this change?

The `is_diag` function had been refactored at some point by @chMoussa and I think the `stride` thing was there because it was especially efficient.

Reply:
I did it actually because I got a weird runtime error with one test in `qadence`. So I refactored this into such simpler form so that it passes.

Reply:
What was the weird error @vytautas-a ?

Reply:
Something about not being able to fit a tensor into designated memory.

Reply:
`FAILED tests/qadence/test_quantum_model.py::test_quantum_model_with_hevo - RuntimeError: setStorage: sizes [15, 16], strides [17, 16], storage offset 2320, and itemsize 8 requiring a storage size of 22392 are out of bounds for storage of size 20480`

Reply:
Hm ok weird indeed, never got this one before.

Reply:
Can you open an issue about this? Just put there a snippet of the previous code, the one you replaced, and the description of the error. Thanks!

Reply:
Raised an issue: #283